gravitee-io/gravitee-access-management
# feat: email verification link registration

## :id: Reference related issue. 

- relates to https://gravitee.atlassian.net/browse/AM-517
- closes:
   - https://gravitee.atlassian.net/browse/AM-446
   - https://gravitee.atlassian.net/browse/AM-518
   - https://gravitee.atlassian.net/browse/AM-424
   - https://gravitee.atlassian.net/browse/AM-423

## :pencil2: A description of the changes proposed in the pull request

This PR proposes the email verification link on self registration

## :memo: Test scenarios 

1) Simple Email verification
- Go to `Settings > Login` (or the Login settings in you application) and enable user registration
- Go to `Settings > User Accounts` under `Security`  (either Domain or Application) and enable `Account Verification via Email` under `User registration` 
- Go you application and initiate the login flow via the link in the `Overview` section
- Sign up to the application. If successful you should have this screen below

<img width="688" alt="image" src="https://github.com/gravitee-io/gravitee-access-management/assets/8531515/f3dca5c8-8d31-476d-8a4c-93b74f1ac4bd">

- Go yo your inbox and click on the verification link

If the verification succeeds you should see:
<img width="657" alt="image" src="https://github.com/gravitee-io/gravitee-access-management/assets/8531515/99d038df-ec06-44b8-8e61-05e1f3cb3f99">

If the link is not longer valid:
<img width="685" alt="image" src="https://github.com/gravitee-io/gravitee-access-management/assets/8531515/ce68bf86-9b8e-4315-8146-8c04203050da">

2) Email verification with Auto login
- Go to `Settings > Login` (or the Login settings in you application) and enable user registration
- Go to `Settings > User Accounts` under `Security`  (either Domain or Application) : 
      - Enable `Account Verification via Email` under `User registration` 
      - Enable Auto Login after User Registration
      
<img width="877" alt="image" src="https://github.com/gravitee-io/gravitee-access-management/assets/8531515/6760368a-8151-4b1f-9d37-fcf133d636c2">

- Go you application and initiate the login flow via the link in the `Overview` section
- Sign up to the application. If successful you should have this screen below

<img width="688" alt="image" src="https://github.com/gravitee-io/gravitee-access-management/assets/8531515/f3dca5c8-8d31-476d-8a4c-93b74f1ac4bd">

- Go yo your inbox and click on the verification link

If the verification succeeds you should be redirected to the Auto Login redirect link.
If the link is not longer valid:
<img width="685" alt="image" src="https://github.com/gravitee-io/gravitee-access-management/assets/8531515/ce68bf86-9b8e-4315-8146-8c04203050da">
